app/assets/javascripts/discourse/components/choose-topic.js.es6

```diff
@@ -32,20 +32,22 @@ export default Ember.Component.extend({
       return;
     }
 
-    searchForTerm(title, { typeFilter: "topic", searchForId: true }).then(
-      function(results) {
-        if (results && results.posts && results.posts.length > 0) {
-          self.set(
-            "topics",
-            results.posts
-              .mapBy("topic")
-              .filter(t => t.get("id") !== currentTopicId)
-          );
-        } else {
-          self.setProperties({ topics: null, loading: false });
-        }
+    searchForTerm(title, {
+      typeFilter: "topic",
+      searchForId: true,
+      restrict_to_archetype: "regular"
```

Comment:
I don't think `searchForTerm` will process the `restrict_to_archetype` key in this object. Also, we're mixing snake_case with camelCase here which can be quite confusing.